openvdb/openvdb/CMakeLists.txt

```diff
@@ -489,6 +538,16 @@ if(OPENVDB_CORE_SHARED)
       PROPERTIES INSTALL_RPATH "${RPATHS}"
     )
   endif()
+
+  if(WIN32)
+    if(NOT CMAKE_MSVC_RUNTIME_LIBRARY)
+      # NOTE: MSVC_RUNTIME_LIBRARY does not propagate to targets, so
+      # also add it explicitly as a compile option
+      set_target_properties(openvdb_shared PROPERTIES
+        MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+      target_compile_options(openvdb_shared PUBLIC "/MD$<$<CONFIG:Debug>:d>")
+    endif()
+  endif()
```

Comment:
@e4lam sorry to drag you into the joy that is CMake and the MSVC runtime, but what do you think of this change? In summary, if `CMAKE_MSVC_RUNTIME_LIBRARY` hasn't been explicitly provided, force the static build of VDB to build with `/MT` and the shared build to build with `/MD` (and their debug variants). Without this, I believe both would build with `/MD` or `/MDd` which they take from the default value of `CMAKE_MSVC_RUNTIME_LIBRARY`. So it's slightly different behaviour for the static build.

Reply:
@Idclip You're right, I did not notice this when I rebuilt OpenVDB ourselves with the quick fix of removing the lines that set CMAKE_MSVC_RUNTIME_LIBRARY to the empty string. Strangely though, it all seems to link fine in Houdini land (eg. the static libUT.lib) and it doesn't seem to cause a dependency in our statically linked executables to the shared CRT. Regardless, I'm all for this change so that the resulting static lib is built with `/MT` and `/MTd`.

The other consideration here is that some people _want_ to compile the static library with `/MD`. ie. they want to statically link in OpenVDB but still link in the shared CRT ([discussion thread](https://www.openvdb.org/forum/?place=msg%2Fopenvdb-forum%2FI5Q_Jn4IGGM%2F8LB0tuuUXdQJ)).  ). But I think that's fine, as long as they can override CMAKE_MSVC_RUNTIME_LIBRARY with the appropriate generator expression.

Reply:
Ah, I think I know why compiling the static openvdb with /MD is not a problem for Houdini. None of our static binaries need to call anything into openvdb, so I suspect maybe the linker is smart enough to not link in openvdb in even though it's specified on the linker line.

Reply:
> The other consideration here is that some people _want_ to compile the static library with `/MD`. ie. they want to statically link in OpenVDB but still link in the shared CRT ([discussion thread](https://www.openvdb.org/forum/?place=msg%2Fopenvdb-forum%2FI5Q_Jn4IGGM%2F8LB0tuuUXdQJ)). But I think that's fine, as long as they can override CMAKE_MSVC_RUNTIME_LIBRARY with the appropriate generator expression.

Absolutely, you can provide `-DCMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$<CONFIG:Debug>:Debug>DLL"` as an argument which will set /MD and /MDd as expected to all artefacts.